ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader_builder.cpp

```diff
@@ -186,3 +186,9 @@ Datadog::UploaderBuilder::build()
 
     return Datadog::Uploader{ output_filename, ddog_exporter };
 }
+
+void
+Datadog::UploaderBuilder::postfork_child()
+{
+    tag_mutex.unlock();
```

Comment:
I see that the sanitizers are already flagging this ([example](https://github.com/DataDog/dd-trace-py/actions/runs/12394585911/job/34598388160?pr=11768#step:5:819)), but this is undefined behavior if the mutex is already unlocked. I think we need to do what the `Datadog::Uploader` already does and lock the mutexes before forking, fork, and then unlock them in the parent and child.

Reply:
Ah, I just saw this.  Right on--thanks

I hate how std::mutex made pthreads _worse_.  The error-checking type pthreads mutex wraps up these issues.